src/main/java/newbank/server/commands/Command.java

```diff
@@ -1,27 +1,94 @@
 package newbank.server.commands;
 
 import newbank.server.CustomerID;
-import newbank.server.exceptions.CommandInvalidSyntaxException;
-import newbank.server.exceptions.RequestNotAllowedException;
+import newbank.server.NewBank;
 
-/** Abstract representation of a command. */
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Abstract representation of a command.
+ */
 public abstract class Command {
-  public abstract String execute() throws CommandInvalidSyntaxException;
+  protected final NewBank bank;
+  protected final String[] tokens;
+  protected final CustomerID customerID;
+  protected List<Supplier<Optional<String>>> responsibilityChain;
+
+  protected Command(NewBank bank, String[] tokens, CustomerID customerID) {
+    this.bank = bank;
+    this.tokens = tokens;
+    this.customerID = customerID;
+  }
+
+  public abstract String execute();
 
   /**
    * @return the command syntax
    */
-  public String getSyntax() {
-    return "";
+  public abstract String getSyntax();
+
+  /**
+   * Apply each of the functions in the responsibility chain. If any of them
+   * return a String (containing a failure message), stop processing.
+   *
+   * @return `Optional`ly either a string containing a failure message, or null.
+   */
+  protected Optional<String> applyResponsibilityChain() {
+    return responsibilityChain.stream()
```

Comment:
Related to the comment below.

So, for example, here: wouldn't this method be much simpler with a simple if check and a string that is null/empty?